### DISCUSSION TITLE
Expanding Batch-Antivirus functionality for general file sorting/recognition

### DISCUSSION THREAD

**Discussion by shodanx2** [anic17/Batch-Antivirus#14] [Ideas]
Hi,

I found this thread and it seems to be something right up your alley

https://old.reddit.com/r/DataHoarder/comments/12b7z18/has_anyone_ever_endeavor_a_wikipedia_of_all_known/jew2z4e/

So, imagine using your program but not just for detecting viruses, but detecting any known file and it's origin.

For instance this dataset 

https://www.nist.gov/itl/ssd/software-quality-group/national-software-reference-library-nsrl/nsrl-download/current-rds

One use I looking for was, to scan entire hard drives and report only files not in the database. 

All operating files and program would be in this database, so it would be a great way to find all user's lost files no matter where they stuffed it !

**Comment by anic17** [comment]
Hi, thanks for your suggestion.

Unfortunately, I don't think it's possible to embed a 91 GB or even a 13 GB database on Batch Antivirus. Furthermore, due to performance issues related to batch, it would take hours to do a search on a file. In order to fix this issue, an SQL reader program would have to be embedded which would mean more external programs, thus being more a pack of applications with a batch launcher. Nonetheless, I appreciate your suggestion, and I'm open to other suggestions.

Thanks for your comment,
anic17
